src/extension/background-script/connectors/lawallet.ts

```diff
@@ -0,0 +1,547 @@
+  urlxPublicKey: string;
+  relayList: string[];
+  relayUrl: string;
+}
+
+export class HttpError extends Error {
+  status: number;
+  error: Error | undefined;
+
+  constructor(status: number, message: string, error?: Error) {
+    super(message);
+    this.status = status;
+    this.error = error;
+  }
+}
+
+export default class LaWallet implements Connector {
+  account: Account;
+  config: Config;
+  relay: Relay;
+  public_key: string;
+  access_token?: string;
+  access_token_created?: number;
+  refresh_token?: string;
+  refresh_token_created?: number;
+  noRetry?: boolean;
+
+  invoices_paid: InvoiceCache = {};
+  last_invoice_check: number = 0;
+
+  constructor(account: Account, config: Config) {
+    this.account = account;
+    this.config = config;
+    this.public_key = new Nostr(config.privateKey).getPublicKey();
+    this.relay = relayInit(config.relayUrl);
+  }
+
+  async init() {
+    return Promise.resolve();
+  }
+
+  unload() {
+    this.relay.close();
+    return Promise.resolve();
+  }
+
+  get supportedMethods() {
+    return [
+      "getInfo",
+      "makeInvoice",
+      "sendPayment",
+      "signMessage",
+      "getInvoices",
+      "getBalance",
+    ];
+  }
+
+  async connectPeer(): Promise<ConnectPeerResponse> {
+    console.error(
+      `${this.constructor.name} does not implement the getInvoices call`
+    );
+    throw new Error("Not yet supported with the currently used account.");
+  }
+
+  private async getInvoices(): Promise<ConnectorTransaction[]> {
+    const transactions: ConnectorTransaction[] = (await this.getTransactions())
+      .data.transactions;
+
+    return transactions.filter(
+      (transaction) => transaction.type === "received"
+    );
+  }
+
+  async getTransactions(): Promise<GetTransactionsResponse> {
+    const _transactions: Event[] = await LaWallet.request(
+      this.config.apiEndpoint,
+      "POST",
+      "/nostr/fetch",
+      {
+        authors: [this.config.ledgerPublicKey, this.config.urlxPublicKey],
+        kinds: [1112],
+        since: 0,
+        "#t": ["internal-transaction-ok", "inbound-transaction-start"],
+        "#p": [this.public_key],
+      }
+    );
+
+    const transactions = _transactions.map((event) => {
+      return {
+        ...event,
+        kind: event.kind as EventKind,
+      };
+    }) as Event[];
+
+    const parsedTransactions: ConnectorTransaction[] = await Promise.all(
+      transactions.map(
+        parseTransaction.bind(this, this.public_key, this.config.privateKey)
+      )
+    );
+
+    return {
+      data: {
+        transactions: parsedTransactions.sort(
+          (a, b) => b.settleDate - a.settleDate
+        ),
+      },
+    };
+  }
+
+  private async getPayments(): Promise<ConnectorTransaction[]> {
+    const transactions: ConnectorTransaction[] = (await this.getTransactions())
+      .data.transactions;
+
+    return transactions.filter((transaction) => transaction.type === "sent");
+  }
+
+  async getInfo(): Promise<
```

Comment:
![image](https://github.com/getAlby/lightning-browser-extension/assets/55848322/20df3137-2c10-4d9f-9d22-bb60ec1559a8)
i still get weird alias did you fixed it?